### ISSUE TITLE
useTransition - startTransition does not work on React.memo when is SimpleMemoComponent

### ISSUE THREAD

**Issue by salvoravida** [facebook/react#17318]
**Do you want to request a *feature* or report a *bug*?**
bug
**What is the current behavior?**
useTransition - startTransition do not work on React.memo when is SimpleMemoComponent

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**

**What is the expected behavior?**
startTransition work with SimpleMemoComponent

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

**Comment by threepointone** [comment]
Please make a reproducing example on codesandbox, it’s not clear what you’re asking.

**Comment by salvoravida** [comment]
Click on "With Transition Start"
OK -> https://codesandbox.io/s/test-redux-in-concurrent-mode-with-react-redux-salvoravidareact-redux-9lun7
KO -> https://codesandbox.io/s/test-redux-in-concurrent-mode-with-react-redux-salvoravidareact-redux-68pbd

only line 85 is changed

```jsx
const _MyComponent = () => {
  const status = useSelector(s => s.status);

  if (status === STATUS.PENDING) {
    throw promise;
  }
  return null;
};

//_MyComponent.defaultProps = {};
const MyComponent = React.memo(_MyComponent);
```

the problem maybe is here in react-dom 
```jsx
function updateMemoComponent(current$$1, workInProgress, Component, nextProps, updateExpirationTime, renderExpirationTime) {
  if (current$$1 === null) {
    var type = Component.type;

    if (isSimpleFunctionComponent(type) && Component.compare === null && // SimpleMemoComponent codepath doesn't resolve outer props either.
    Component.defaultProps === undefined) {
      var resolvedType = type;

      {
        resolvedType = resolveFunctionForHotReloading(type);
      } // If this is a plain function component without default props,
      // and with only the default shallow comparison, we upgrade it
      // to a SimpleMemoComponent to allow fast path updates.


      workInProgress.tag = SimpleMemoComponent;
      workInProgress.type = resolvedType;

      {
        validateFunctionComponentInDev(workInProgress, type);
      }

      return updateSimpleMemoComponent(current$$1, workInProgress, resolvedType, nextProps, updateExpirationTime, renderExpirationTime);
    }
```

**Comment by dai-shi** [comment]
@salvoravida Hi, do you think you can create a smaller reproduction? You don't need redux or even useTransition to reproduce this bug, do you? If what you are observing is correct, this issue is related with https://github.com/facebook/react/issues/17314.

**Comment by gaearon** [comment]
We haven't checked if this is a bug. It might be.

**Comment by dai-shi** [comment]
@gaearon _Probably_, this issue and #17314 can be merged into #17028. All of them are originated from my code. #17028 has a small reproduction, while this issue is good because it points out the issue only happens with React.memo only when it has no `areEqual`.

**Comment by acdlite** [comment]
Thanks for the bug report!

I believe this was fixed by https://github.com/facebook/react/pull/18091

You can confirm using `react@0.0.0-experimental-ea6ed3dbb` and `react-dom@react@0.0.0-experimental-ea6ed3dbb`.

I'm going to close this issue, but if the bug persists please comment and we'll reopen.

**Comment by dai-shi** [comment]
Confirmed the fix with `0.0.0-experimental-5de5b6150` on my end. Thanks so much!

**Comment by salvoravida** [comment]
great! thanks!
